functions/backup.bash

```diff
@@ -1,228 +1,341 @@
-      chown "${backupuser}":backup "${storage}"/slots/slot${counter}
-    else
-      su - "${backupuser}" -c "/usr/sbin/amlabel ${config} ${config}-${counter} slot ${counter}"
-    fi
-    ((counter += 1))
-  done
+  if cond_redirect chsh --shell /bin/bash "$backupUser"; then echo "OK"; else echo "FAILED (chsh ${backupUser})"; return 1; fi
 
-  # create cronjob to save copies of the Amanda database
-  echo "0 1 * * * ${backupuser} /usr/sbin/amdump ${config} &> /dev/null" > /etc/cron.d/amanda
-  echo "0 18 * * * ${backupuser} /usr/sbin/amcheck -m ${config} &> /dev/null" >> /etc/cron.d/amanda
-  if [ "${tapetype}" = "DIRECTORY" ]; then
-    mkdir -p "${storage}"/amanda-backups
-    chown "${backupuser}":backup "${storage}"/amanda-backups
-    echo "0 2 * * * root (cd /; /bin/tar czf ${storage}/amanda-backups/amanda_data_\$(date +\\%Y\\%m\\%d\\%H\\%M\\%S).tar.gz etc/amanda var/lib/amanda var/log/amanda; find ${storage} -name amanda_data_\\* -mtime +30 -delete) &> /dev/null" >> /etc/cron.d/amanda
+  if ! [[ $(dpkg -s 'amanda-common' 'amanda-server' 'amanda-client') ]]; then
+    echo -n "$(timestamp) [openHABian] Installing Amanda backup system... "
+    if cond_redirect apt-get install --yes amanda-common amanda-server amanda-client; then echo "OK"; else echo "FAILED"; return 1; fi
   fi
 }
 
+## Setup a Amanda based backup using local storage, AWS, or a set of SD cards.
+##
+##    amanda_setup()
+##
 amanda_setup() {
-  local matched
-  local canceled
-  local backupuser="backup"
-  local config=${storageconfig:-openhab-dir}
-  local tapes=${storagetapes:-15}
-  local size=${storagecapacity:-1024}
-  local dir=${storagedir:-1024}
-  local querytext="So you are about to install the Amanda backup solution.\\nDocumentation is available at the previous openHABian menu point,\\nat /opt/openhabian/docs/openhabian-amanda.md or at https://github.com/openhab/openhabian/blob/master/docs/openhabian-amanda.md\\nHave you read this document ?"
-  local introtext="This will setup a backup mechanism to allow for saving your openHAB setup and modifications to either USB attached or Amazon cloud storage.\\nYou can add your own files/directories to be backed up, and you can store and create clones of your openHABian SD card to have an all-ready replacement in case of card failures."
-  local successtext="Setup was successful. Amanda backup tool is now taking backups at 01:00. For further readings, start at http://wiki.zmanda.com/index.php/User_documentation."
-
- 
-  # shellcheck disable=SC2154
-  if [[ -z $INTERACTIVE ]] && [[ -z "$backupdrive" ]]; then return 0; fi
-
-  if [[ -n $INTERACTIVE ]]; then
-    if ! (whiptail --title "Amanda backup installation" --yes-button "Yes" --no-button "No, I'll go read it" --defaultno --yesno "$querytext" 10 80); then return 0; fi
+  if [[ -z $INTERACTIVE ]]; then
+    echo "$(timestamp) [openHABian] Amanda backup setup must be run in interactive mode! Canceling Amanda backup setup!"
```

Comment:
No, remember I talked about making the `amanda_install()` function to allow it to be called from uninteractive and then use the `create_amanda_config` inside of setup_mirrior_SD to complete the setup. look at it again.

Reply:
So which of those question is "no" your answer on? if you rebased ?
I remember we talked about but IIRC the outcome was that I asked you to take my most recent Amanda code and eventually standardize/rewrite it again. BTW it already implements both, interactive AND unattended.
Current code calls amanda_setup() from unattended openhabian-setup (and that your PR does not change).

Reply:
Yes I rebased, no it should not be called in unattended.

Yes it does, It calls it from you mirror sd setup function. I missed removing the call in openhabian-setup.sh but it has since been changed and corrected. look closely.

Reply:
ok, I checked again and yes it looks almost fine now.
However I would still like amanda_setup not to be called nested from setup_mirror_SD but from main script (albeit this time *after* setup_mirror_SD).
One reason is I really dislike nesting in general as it potentially fogs dependencies.
The other is you can select setup_mirror_SD from the 5X menu _without_ that you want to setup Amanda. Current PR code would do that though.

Reply:
OK, `amanda_setup` is not called in the `setup_mirror_SD` but the install function `amanda_install` is. I would like to keep it where it is because the `create_amanda_config` is called in the `setup_mirror_SD` right after `amanda_install`. The way I have it set up makes it fairly clear about dependencies and moving it outside would make it so that the config would not be created.

Being able to install SD mirroring without Amanda was intentional in an interactive setup as the option to install Amanda is a separate option and it allows more finite control over the system and what is set up. I think it would be better to simply recommend to use the leftover space to setup Amanda after mirroring setup is finished. Then Amanda can also be setup to have proper mailing support as well.

Reply:
Ok then. Not my preference but I don't mind to keep it as-is.